clang/lib/Lex/InitHeaderSearch.cpp

```diff
@@ -339,13 +340,11 @@ void InitHeaderSearch::AddDefaultIncludePaths(
   if (triple.isOSDarwin()) {
```

Comment:
Do you think we can just clean up this code block by moving all logics to driver?

Reply:
Looks like that was previously attempted by @ldionne  in https://github.com/llvm/llvm-project/pull/75841 

The problem was appending `/Library/Frameworks` to the sysroot. I'll try rebasing that.

Reply:
I looked into this a bit more. IMO the only way to move all this code to the driver is to either 
 
1. changing search path order between Framework search paths & other default search paths like `-internal-isystem $(SDKROOT)/usr/include` (since existing `iframework` puts the search path at the top instead of at the bottom like what `InitHeaderSearch` does)
OR 
2. introducing some kind of search path argument to preserve back precedence for S/L/F e.g. a  `-internal-iframework`

Both feel out of scope to me.

Reply:
FWIW I agree with your assessment @cyndyishida. I just did the exercise of picking up #75841 and we basically need to pass `-iframework` from the driver to the frontend, but if we limit ourselves to existing functionality in the driver that ends up putting the framework search paths before other system headers. I think that should be attempted as a separate patch that tries reviving #75841.

Reply:
I'm trying to revive this as https://github.com/llvm/llvm-project/pull/120149